### PR TITLE
This PR adds the UI of Mess Section: Feedback, Complaint, MOM, Tenders and their integration with the backend

### PR DESCRIPTION
**Pull request by mtabishk**
UI and integration of the Mess Section: 
1. Feedback
2. Complaint
3. MOM
4. Tenders

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,8 +0,0 @@
-DJANGO_SECRET_KEY=secret key
```
Comment (ashutoshc8101) [inline .env.example:1]: Revert this change.

File: api/authentication/backends.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
 # Take environment variables from .env file
 environ.Env.read_env(os.path.join(BASE_DIR.parent, '.env'))
 
+
```
Comment (ashutoshc8101) [inline api/authentication/backends.py:21]: Changes to this file are not required.

File: app/.env.symlink [hunk 3]
```diff
@@ -1 +0,0 @@
-../.env
```
Comment (ashutoshc8101) [inline app/.env.symlink:1]: This change is not required.

File: app/.gitignore [hunk 4]
```diff
@@ -43,4 +43,4 @@ app.*.map.json
 # Android Studio will place build artifacts here
 /android/app/debug
 /android/app/profile
-/android/app/release
+/android/app/release
```
Comment (ashutoshc8101) [inline app/.gitignore:46]: Revert this change.

File: app/android/app/build.gradle [hunk 5]
```diff
@@ -52,10 +52,10 @@ android {
...
     signingConfigs {
         debug {
-            storeFile file('key.jks')
-            storePassword 'instituteapp'
-            keyAlias 'androiddebugkey'
-            keyPassword 'instituteapp'
+            storeFile file('upload-keystore.jks')
+            storePassword 'password'
+            keyAlias 'uploadkeystore'
+            keyPassword 'password'
```
Comment (ashutoshc8101) [inline app/android/app/build.gradle:58]: Don't include this change in the PR as this will break authentication for other devs.

File: app/lib/screens/dashboard/dashboard.dart [hunk 6]
```diff
@@ -64,6 +65,7 @@ class _DashboardState extends State<Dashboard> {
   @override
   void initState() {
     super.initState();
+    //TODO: REMOVE COMMENT
```
Comment (ashutoshc8101) [inline app/lib/screens/dashboard/dashboard.dart:68]: Remove this comment.

File: app/lib/screens/mess/manager/complaints_mess_manager.dart [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+                                                              shape:
+                                                                  CircleBorder(),
+                                                            ),
+                                                          ),
+                                                        ),
+                                                ],
+                                              ),
+                                              Builder(builder: (context) {
+                                                return Text(
+                                                    "1B Mess | Breakfast | 06-07-2022",
```
Comment (ashutoshc8101) [inline app/lib/screens/mess/manager/complaints_mess_manager.dart:207]: Why is this text hardcoded? Shouldn't there be data fetched from backend?

File: app/lib/screens/mess/manager/complaints_mess_manager.dart [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+                                              }),
+                                            ],
+                                          ),
+                                        ),
+                                      ),
+                                      Padding(
+                                        padding: const EdgeInsets.all(8.0),
+                                        child: Wrap(
+                                          children: [
+                                            Text("jkwheqk " * 30,
```
Comment (ashutoshc8101) [inline app/lib/screens/mess/manager/complaints_mess_manager.dart:220]: Didn't understand the purpose of this change.

File: app/lib/screens/mess/manager/mess_manager.dart [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+    "Mess MOM",
+  ];
+
+  final List<Widget> _messRoutes = [
+    FeedbackMessManager(),
+    ComplaintMessManager(),
+    MessMenu(),
+    // MessLeaveMess(),
+    // FoodRequestMess(),
+    // MessMOMMess(),
```
Comment (ashutoshc8101) [inline app/lib/screens/mess/manager/mess_manager.dart:65]: Remove these comments.

File: app/lib/screens/mess/student/complains_mess_student.dart [hunk 10]
```diff
@@ -0,0 +1,373 @@
...
+
+  @override
+  State<ComplaintsMess> createState() => _ComplaintsMessState();
+}
+
+class _ComplaintsMessState extends State<ComplaintsMess> {
+  List<String> _messFill = ["Fill as Anonymous", "Use your credentials"];
+  String _messFillValue = "Fill as Anonymous";
+  List<String> _mess = ["1 B Mess", "120 Mess", "Girls Mess"];
+  String _messValue = "1 B Mess";
```
Comment (ashutoshc8101) [inline app/lib/screens/mess/student/complains_mess_student.dart:22]: I guess this data should be from API.

File: app/lib/screens/mess/student/complains_mess_student.dart [hunk 11]
```diff
@@ -0,0 +1,373 @@
...
+                        const SizedBox(
+                          height: 20.0,
+                        ),
+                        Padding(
+                          padding: EdgeInsets.symmetric(horizontal: 40.0),
+                          child: TextField(
+                            maxLines: 5,
+                            decoration: InputDecoration(
+                              border: OutlineInputBorder(
+                                  //borderRadius: BorderRadius.circular(10.0),
```
Comment (ashutoshc8101) [inline app/lib/screens/mess/student/complains_mess_student.dart:257]: Avoid residual comments.

File: app/lib/screens/mess/student/mess_student.dart [hunk 12]
```diff
@@ -0,0 +1,273 @@
...
+          ),
+        ],
+      ),
+    );
+  }
+}
+    // return FutureBuilder(
+    //     future: fetchMessData(),
+    //     builder: (BuildContext context,
+    //         AsyncSnapshot<Map<String, List<Mess>>> snapshot) {
```
Comment (ashutoshc8101) [inline app/lib/screens/mess/student/mess_student.dart:178]: Ditto

File: app/lib/services/auth.service.dart [hunk 13]
```diff
@@ -25,6 +25,7 @@ class AuthService with ChangeNotifier {
 
   AuthService() {
     _googleSignIn.onCurrentUserChanged.listen((GoogleSignInAccount? account) {
+      print(account);
```
Comment (ashutoshc8101) [inline app/lib/services/auth.service.dart:28]: Revert this change.

**Review by ashutoshc8101 — CHANGES_REQUESTED** [review]
Left few comments.

File: app/lib/screens/mess/student/complains_mess_student.dart (hunk 10 above)
Comment (mtabishk) [reply app/lib/screens/mess/student/complains_mess_student.dart:22]: There is no API yet for this data

File: app/.gitignore (hunk 4 above)
Comment (mtabishk) [reply app/.gitignore:46]: There is no change here

File: .env.example (hunk 1 above)
Comment (mtabishk) [reply .env.example:1]: Done

**Review by ashutoshc8101 — APPROVED** [review]
